.github/workflows/macos.yaml

```diff
@@ -180,6 +184,7 @@ jobs:
         ARTIFACTORY_URL: datadog.jfrog.io
         ARTIFACTORY_GEMS_PATH: artifactory/api/gems/agent-gems
         ARTIFACTORY_PYPI_PATH: artifactory/api/pypi/agent-pypi/simple
+        INTEGRATIONS_CORE_VERSION: ${{ github.event.inputs.integrations_core_ref || env.DEFAULT_INTEGRATIONS_CORE_REF }}
```

Comment:
I believe this breaks Agent releases.

In the `agent.omnibus-build` invoke task, the logic related to `INTEGRATIONS_CORE_VERSION` is defined [here](https://github.com/DataDog/datadog-agent/blob/17ac27f0794404b16932d444ea43d6b88445b0dc/tasks/agent.py#L549-L564):

- load the `INTEGRATIONS_CORE_VERSION` entry from the `release.json` file,
- if the `INTEGRATIONS_CORE_VERSION` environment variable exists and is non-empty, use it instead.

The result, with this change, is that the contents of the `release.json` file are always ignored, since the `INTEGRATIONS_CORE_VERSION` variable is never empty. This means that release builds will use `master` of `integrations-core`, which is not the desired result.

Reply:
Sorry I missed the top of the method.
So the expected behaviour would be:
- either we have an input from github (so that `integrations-core` can test a dedicated branch)
- or we take what comes from `release.json`, using the `RELEASE_VERSION` like in [here](https://github.com/DataDog/datadog-agent/blob/17ac27f0794404b16932d444ea43d6b88445b0dc/tasks/utils.py#L465-L472)
- or we take default
@KSerrania do I forgot something?

Reply:
@KSerrania this version should be better, thanks in advance for your review